static/app/views/dashboardsV2/widget/metricWidget/index.tsx

```diff
@@ -290,57 +251,37 @@ class MetricWidget extends AsyncView<Props, State> {
                     project={project}
                     widget={{
                       title,
-                      queries,
-                      yAxis: metric?.name,
+                      searchQuery,
+                      displayType,
+                      groupings: queries,
                     }}
                   />
                 </VisualizationWrapper>
               </BuildStep>
               <ChooseDataSetStep value={DataSet.METRICS} onChange={onChangeDataSet} />
               <BuildStep
-                title={t('Choose your y-axis metric')}
-                description={t('Determine what type of metric you want to graph by.')}
+                title={t('Begin your search')}
+                description={t('Select a query to compare projects, tags, etc.')}
               >
-                <StyledSelectField
-                  name="metric"
-                  choices={metrics.map(m => [m, m.name])}
-                  placeholder={t('Select metric')}
-                  onChange={this.handleMetricChange}
-                  value={metric}
-                  components={{
-                    Option: ({
-                      label,
-                      ...optionProps
-                    }: OptionProps<{
-                      label: string;
-                      value: string;
-                    }>) => {
-                      const {selectProps} = optionProps;
-                      const {inputValue} = selectProps;
-
-                      return (
-                        <components.Option label={label} {...optionProps}>
-                          <Highlight text={inputValue ?? ''}>{label}</Highlight>
-                        </components.Option>
-                      );
-                    },
-                  }}
-                  inline={false}
-                  flexibleControlStateSize
-                  stacked
-                  allowClear
+                <SearchQueryField
+                  api={this.api}
+                  tags={metricTags}
+                  orgSlug={orgSlug}
+                  projectSlug={project.slug}
+                  query={searchQuery}
+                  onSearch={newQuery => this.handleFieldChange('searchQuery', newQuery)}
+                  onBlur={newQuery => this.handleFieldChange('searchQuery', newQuery)}
                 />
               </BuildStep>
               <BuildStep
```

Comment:
If I add 3 groupings and delete the second one, it deletes both the second and third- is that expected behaviour?

Reply:
I copied the delete query method from here https://github.com/getsentry/sentry/blob/200928c6914f34db7b9280b6119d02502f0cce46/static/app/components/modals/addDashboardWidgetModal.tsx#L281 and did not notice this error... I have just fixed it...Thanks!

Reply:
@shruthilayaj @priscilawebdev This is also a bug for Dashboards. I've addressed it here https://github.com/getsentry/sentry/pull/25724

Reply:
@priscilawebdev it might be worth having a test for this; it doesn't have to be in this PR.

Reply:
yes sure... I will write tests before removing the feature flag